app/assets/javascripts/discourse/controllers/bookmark.js.es6

```diff
@@ -0,0 +1,224 @@
+        date: this.laterToday().format(I18n.t("dates.time"))
+      })
+    );
+  },
+
+  @discourseComputed()
+  tomorrowFormatted() {
+    return htmlSafe(
+      I18n.t("bookmarks.reminders.tomorrow", {
+        date: this.tomorrow().format(I18n.t("dates.time_short_day"))
+      })
+    );
+  },
+
+  @discourseComputed()
+  nextBusinessDayFormatted() {
+    return htmlSafe(
+      I18n.t("bookmarks.reminders.next_business_day", {
+        date: this.nextBusinessDay().format(I18n.t("dates.time_short_day"))
+      })
+    );
+  },
+
+  @discourseComputed()
+  nextWeekFormatted() {
+    return htmlSafe(
+      I18n.t("bookmarks.reminders.next_week", {
+        date: this.nextWeek().format(I18n.t("dates.month_day_time"))
+      })
+    );
+  },
+
+  @discourseComputed()
+  nextMonthFormatted() {
+    return htmlSafe(
+      I18n.t("bookmarks.reminders.next_month", {
+        date: this.nextMonth().format(I18n.t("dates.month_day_time"))
+      })
+    );
+  },
+
+  @discourseComputed()
+  userHasTimezoneSet() {
+    return !_.isEmpty(this.userTimezone());
+  },
+
+  saveBookmark() {
+    const reminderAt = this.reminderAt();
+    const data = {
+      reminder_type: this.selectedReminderType,
+      reminder_at: reminderAt ? reminderAt.toISOString() : null,
+      name: this.name,
+      post_id: this.model.postId
+    };
+
+    return ajax("/bookmarks", { type: "POST", data: data });
+  },
+
+  reminderAt() {
+    if (this.selectedReminderType === null) {
+      return;
+    }
+
+    switch (this.selectedReminderType) {
+      case REMINDER_TYPES.AT_DESKTOP:
+        // TODO: Implement at desktop bookmark reminder functionality
+        return "";
+      case REMINDER_TYPES.LATER_TODAY:
+        return this.laterToday();
+      case REMINDER_TYPES.NEXT_BUSINESS_DAY:
+        return this.nextBusinessDay();
+      case REMINDER_TYPES.TOMORROW:
+        return this.tomorrow();
+      case REMINDER_TYPES.NEXT_WEEK:
+        return this.nextWeek();
+      case REMINDER_TYPES.NEXT_MONTH:
+        return this.nextMonth();
+      case REMINDER_TYPES.CUSTOM:
+        // TODO: Implement custom bookmark reminder times
+        return "";
+    }
+  },
+
+  nextWeek() {
+    return this.startOfDay(this.now().add(7, "days"));
+  },
+
+  nextMonth() {
+    return this.startOfDay(this.now().add(1, "month"));
+  },
+
+  nextBusinessDay() {
+    const currentDay = this.now().isoWeekday(); // 1=Mon, 7=Sun
+    let next = null;
+
+    // friday
+    if (currentDay === 5) {
+      next = this.now().add(3, "days");
+      // saturday
+    } else if (currentDay === 6) {
+      next = this.now().add(2, "days");
+    } else {
+      next = this.now().add(1, "day");
+    }
+
+    return this.startOfDay(next);
+  },
+
+  tomorrow() {
+    return this.startOfDay(this.now().add(1, "day"));
+  },
+
+  startOfDay(momentDate) {
+    return momentDate.hour(START_OF_DAY_HOUR).startOf("hour");
+  },
+
+  userTimezone() {
+    return this.currentUser.timezone;
```

Comment:
are we always sure we have a currentUser here ?

Reply:
Yes, you cannot open the bookmark modal (and the old bookmark button does not work either) unless you have a current user. Will just double check that I can't get to it anyway